### PR TITLE
chore(deps): bump i18next libraries

### PR DESCRIPTION
**Pull request by ckniffen**
- i18next: ^10.6.0 => ^21.6.16
- i18next-browser-languagedetector: ^2.1.1 => ^6.1.4
- i18next-xhr-backend@^1.5.1 => i18next-http-backend@^1.4.0
- react-i18next": ^9.0.3 => ^11.16.7

## High Level Overview of Change

- translate was changed to withTranslation
- useTranslation hook was added

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: src/i18n.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import i18n from 'i18next';
-import XHR from 'i18next-xhr-backend';
-import languageDetector from './languageDetector';
+import Backend from 'i18next-http-backend';
+import LanguageDetector from 'i18next-browser-languagedetector';
 
 const options = {
-  debug: process.env.NODE_ENV === 'development',
+  debug: true,
```
Comment (mvadari) [inline src/i18n.js:6]: Why was this changed?

**Review by mvadari — APPROVED** [review]
lgtm, assuming you checked to make sure the translations were actually working

File: src/i18n.js (hunk 1 above)
Comment (ckniffen) [reply src/i18n.js:6]: Great catch.

**Comment by ckniffen** [comment]
> lgtm, assuming you checked to make sure the translations were actually working

If I re-enable the picker it works great.

File: src/containers/Token/TokenTransactionsTable/index.js [hunk 2]
```diff
@@ -69,7 +70,7 @@ export const TokenTxTable = props => {
               </div>
             </div>
             <div className={`col-type tx-type ${tx.type}`}>
-              <TxLabel type={tx.type} t={t} />
+              <TxLabel type={tx.type} />
```
Comment (ledhed2222) [inline src/containers/Token/TokenTransactionsTable/index.js:73]: how come we don't need to pass `t` as a prop anymore? I'm not seeing it used elsewhere here

File: src/index.js [hunk 3]
```diff
@@ -33,15 +33,17 @@ if (!Promise) {
...
-    <I18nextProvider i18n={i18n}>
-      <Provider store={store}>
-        <PersistGate loading={null} persistor={persistor}>
-          <Router>
-            <App />
-          </Router>
-        </PersistGate>
-      </Provider>
-    </I18nextProvider>,
+    <Suspense fallback="Loading">
```
Comment (ledhed2222) [inline src/index.js:36]: nice - just looked up `Suspense`. It has quite an ominous name, but seems very cool

File: src/containers/Token/TokenTransactionsTable/index.js (hunk 2 above)
Comment (ckniffen) [reply src/containers/Token/TokenTransactionsTable/index.js:73]: I updated TxLabel to use the `useTranslation` hook so it wont need to be passed.  I will have more PRs in the future updating more places.  That hook was added in the major version of react-i18next we are updating to.

File: src/index.js (hunk 3 above)
Comment (ckniffen) [reply src/index.js:36]: Yeah.  I still need to research more.  Adding it prevents the page from rendering until the translations are loaded.  Otherwise a double load of the app happens

File: src/containers/Tokens/TokensTable/index.js [hunk 4]
```diff
@@ -72,15 +71,19 @@ const TokensTable = props => {
           )}
         </td>
         <td className="obligations">
-          {tokenInfo.obligations ? processBigNumber(tokenInfo.obligations, lng) : 0}
+          {tokenInfo.obligations
+            ? processBigNumber(tokenInfo.obligations, i18n.resolvedLanguage)
```
Comment (JST5000) [inline src/containers/Tokens/TokensTable/index.js:75]: Why is `i18n` used to replace `lng` on line 52, but `i18n.resolvedLanguages` used instead of `lng` here on line 75? 
(What did `lng` represent?)

File: src/containers/shared/components/Sequence.tsx [hunk 5]
```diff
@@ -31,19 +31,17 @@ const Sequence = (props: Props) => {
...
 Sequence.defaultProps = {
-  t: (d: string) => d,
   sequence: PropTypes.number,
   ticketSequence: 0,
   addContextHelp: false,
   account: '',
 };
 
-export default translate()(Sequence);
+export default Sequence;
```
Comment (JST5000) [inline src/containers/shared/components/Sequence.tsx:47]: Why is Sequence no longer being translated?

Comment (ckniffen) [reply src/containers/shared/components/Sequence.tsx:47]: `translate(...)` populated the `t` prop.  Now `useTranslation()` is used which makes t available via a hook.

File: src/containers/Tokens/TokensTable/index.js (hunk 4 above)
Comment (ckniffen) [reply src/containers/Tokens/TokensTable/index.js:75]: `lng` represents the language code.  Good catch.

**Review by JST5000 — APPROVED** [review]
Once you fix that one `lng` replacement, LGTM :)
